### PR TITLE
Improve errors around following studios

### PR DESCRIPTION
**Pull request by paulkaplan**
Replace the `Error mutating: PERMISSION` with the real ValidationMessage component, appropriately hide it when clicked away and add translation strings specific to "permission" (confirm your email) and a generic "something went wrong following the studio".

### CODE REVIEW

File: src/views/studio/l10n.json [hunk 1]
```diff
@@ -18,6 +18,15 @@
     "studio.updateErrors.thumbnailTooLarge": "Maximum file size is 512 KB and less than 500x500 pixels.",
     "studio.updateErrors.thumbnailInvalid": "Upload a valid image. The file you uploaded was either not an image or a corrupted image.",
 
+    "studio.followErrors.confirmEmail": "Please confirm your email address first",
+    "studio.followErrors.generic": "Something went wrong following the studio",
+
+    "studio.sectionLoadError.projectsHeadline": "Something went wrong loading projects",
+    "studio.sectionLoadError.curatorsHeadline": "Something went wrong loading curators",
+    "studio.sectionLoadError.managersHeadline": "Something went wrong loading managers",
+    "studio.sectionLoadError.activityHeadline": "Something went wrong loading managers",
```
Comment (ericrosenbaum) [inline src/views/studio/l10n.json:27]: look like this is a copy/paste error?

**Comment by ericrosenbaum** [comment]
I was able to test locally by loading a studio page and using offline mode and see all the errors. One thing that seemed odd was that the comments tab has no corresponding error (it was just blank). Maybe that needs to be handled separately?

**Review by ericrosenbaum — APPROVED** [review]
LGTM pending a fix to l10n string copy/paste error, and question about the comments tab.

**Comment by RedGuy12** [comment]
Will following studios actually do something in the update?
________________________________
From: Paul Kaplan ***@***.***>
Sent: Friday, June 18, 2021 10:30:44 AM
To: LLK/scratch-www ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [LLK/scratch-www] Improve errors around following studios (#5633)


Replace the Error mutating: PERMISSION with the real ValidationMessage component, appropriately hide it when clicked away and add translation strings specific to "permission" (confirm your email) and a generic "something went wrong following the studio".

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/LLK/scratch-www/pull/5633

Commit Summary

  *   Include react-hooks eslint plugin to prevent hooks usage errors.
  *   Replace debug error on StudioFollow

File Changes

  *   M package-lock.json<https://github.com/LLK/scratch-www/pull/5633/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519> (6)
  *   M package.json<https://github.com/LLK/scratch-www/pull/5633/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519> (1)
  *   M src/.eslintrc.js<https://github.com/LLK/scratch-www/pull/5633/files#diff-b3d81df455893664810b193106f3543c920c80f9f4281b5541b949a8e1790769> (5)
  *   M src/views/studio/l10n.json<https://github.com/LLK/scratch-www/pull/5633/files#diff-ab077c03911dfa432a252b595e7a140ed24c484823b3d1c089ef48094491527a> (3)
  *   M src/views/studio/studio-follow.jsx<https://github.com/LLK/scratch-www/pull/5633/files#diff-53541cfe159a081b2bd338962a2bd542506d6f13f07234e5d6c2a38bbe7986da> (40)
  *   M src/views/studio/studio.scss<https://github.com/LLK/scratch-www/pull/5633/files#diff-cdf060ddb78289397e0b3fafc16616c1f380b93ad3bc88221eda7b721c7c0f96> (2)

Patch Links:

  *   https://github.com/LLK/scratch-www/pull/5633.patch
  *   https://github.com/LLK/scratch-www/pull/5633.diff

**Comment by paulkaplan** [comment]
@RedGuy12 I'm not sure what you mean or if you've looked at the code. This changes the error you see when you are prevented from following a studio because you have an unconfirmed email address. Following a studio will continue to do what it currently does; it allows you to customize what projects you see on your homepage in the `Projects in Studios I'm Following` row.

**Comment by paulkaplan** [comment]
@ericrosenbaum yeah i think the comment error has to be handled differently, there is no `error` object coming out of the reducer it seems. the other 3 are all handled by the same redux module so it is easy.

**Comment by paulkaplan** [comment]
(also the comments one handles the issue of viewing deleted studios more gracefully so i don't think it is needed)

**Comment by RedGuy12** [comment]
> Following a studio will continue to do what it currently does; it allows you to customize what projects you see on your homepage in the `Projects in Studios I'm Following` row.

Oh, I thought following studios was supposed to give you notifications for studio activity, but it was broken. I was asking if that would be fixed. But it seems like I was mistaken. My bad!

**Pull request merged**